### PR TITLE
[Dashboard] Replace JS and TS icons

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on replacing icon imports for `JavaScript` and `TypeScript` with custom components, enhancing the visual representation in various components.

### Detailed summary
- Replaced `SiJavascript` with `JavaScriptIcon` in:
  - `ConnectPlaygroundButton.tsx`
  - `CodeOptionButton.tsx`
  - `CodeSegment.tsx`
- Replaced `SiTypescript` with `TypeScriptIcon` in:
  - `SupportedPlatformLink.tsx`
  - `ConnectSDKCard.tsx`
  - `CodeSegment.tsx`
- Added new components:
  - `JavaScriptIcon.tsx`
  - `TypeScriptIcon.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4812?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4812** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4812?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4811** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4811?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4810** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/components/icons/brand-icons/JavaScriptIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      xmlns="http://www.w3.org/2000/svg"
+      fill="currentColor"
+      width={24}
+      height={24}
+      {...props}
+    >
+      <title>JavaScript</title>
+      <path d="M0 0h24v24H0V0zm22.034 18.276c-.175-1.095-.888-2.015-3.003-2.873-.736-.345-1.554-.585-1.797-1.14-.091-.33-.105-.51-.046-.705.15-.646.915-.84 1.515-.66.39.12.75.42.976.9 1.034-.676 1.034-.676 1.755-1.125-.27-.42-.404-.601-.586-.78-.63-.705-1.469-1.065-2.834-1.034l-.705.089c-.676.165-1.32.525-1.71 1.005-1.14 1.291-.811 3.541.569 4.471 1.365 1.02 3.361 1.244 3.616 2.205.24 1.17-.87 1.545-1.966 1.41-.811-.18-1.26-.586-1.755-1.336l-1.83 1.051c.21.48.45.689.81 1.109 1.74 1.756 6.09 1.666 6.871-1.004.029-.09.24-.705.074-1.65l.046.067zm-8.983-7.245h-2.248c0 1.938-.009 3.864-.009 5.805 0 1.232.063 2.363-.138 2.711-.33.689-1.18.601-1.566.48-.396-.196-.597-.466-.83-.855-.063-.105-.11-.196-.127-.196l-1.825 1.125c.305.63.75 1.172 1.324 1.517.855.51 2.004.675 3.207.405.783-.226 1.458-.691 1.811-1.411.51-.93.402-2.07.397-3.346.012-2.054 0-4.109 0-6.179l.004-.056z" />
+    </svg>
+  );
```
Comment (MananTank) [inline apps/dashboard/src/components/icons/brand-icons/JavaScriptIcon.tsx:17]: Name the component - JavaScriptIcon instead

File: apps/dashboard/src/components/icons/brand-icons/TypeScriptIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    </svg>
+  );
+};
```
Comment (MananTank) [inline apps/dashboard/src/components/icons/brand-icons/TypeScriptIcon.tsx:18]: <i class='graphite__hidden'>[Re: line +19]<br/><br/></i>

Name the component TypeScriptIcon instead<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4812?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by MananTank — CHANGES_REQUESTED** [review]
address comments

**Review by MananTank — APPROVED** [review]

**Pull request merged**